python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -0,0 +1,421 @@
+        return iter(
+            [
+                self.All,
+                self.StateTransition,
+                self.MonitorState,
+                self.MonitorHeartBeat,
+                self.Setup,
+                self.Configure,
+                self.TakeFlat,
+                self.DryTest,
+            ]
+        )
+
+
+class ATCalsys(BaseCalsys):
+    """LSST Auxiliary Telescope Calibration System.
+
+    ATCalsys encapsulates core functionality from the following CSCs: ATCamera,
+    ATWhiteLight, ATMonochromoter, Fiberspectrograph, Electrometer.
+
+    Parameters
+    ----------
+    domain : `lsst.ts.salobj.Domain`
+        Domain for remotes. If `None` create a domain.
+    log : `logging.Logger`
+        Optional logging class to be used for logging operations. If `None`,
+        creates a new logger. Useful to use in salobj.BaseScript and allow
+        logging in the class use the script logging.
+    intended_usage: `int`
+        Optional integer that maps to a list of intended operations. This is
+        used to limit the resources allocated by the class by gathering some
+        knowledge about the usage intention. By default allocates all
+        resources.
+    """
+
+    def __init__(
+        self,
+        domain: typing.Optional[salobj.Domain] = None,
+        log: typing.Optional[logging.Logger] = None,
+        intended_usage: typing.Optional[int] = None,
+    ) -> None:
+
+        self.electrometer_index = 1
+        self.fiberspectrograph_index = 1
+
+        super().__init__(
+            components=[
+                "ATCamera",  # isn't this just LATISS?
+                "ATMonochromator",
+                "ATWhiteLight",
+                f"FiberSpectrograph:{self.fiberspectrograph_index}",
+                f"Electrometer:{self.electrometer_index}",
+            ],
+            instrument_setup_attributes=["filter", "calib_type"],
+            domain=domain,
+            log=log,
+            intended_usage=intended_usage,
+        )
+
+        self.entrance_slit = 0.5
+        self.exit_slit = 0.5
+
+        # White lamp config
+        self.timeout_lamp_warm_up = 60 * 15
+        self.track_lamp_warmup = 60
+        self.cmd_timeout = 30
+        self.timeout_std = 20
+        self.whitelight_startup_power = 1000
+
+        # Chiller config
+        self.timeout_chiller_cool_down = 60 * 15
+        self.chiller_temp_tolerance_relative = 0.2
+        self.chiller_temperature = 20
+
+        self.filter_gratings = {
+            "u": 0,  # "Blue"
+            "g": 0,
+            "r": 0,
+            "i": 1,  # "Red"
+            "z": 1,
+            "y": 1,
+        }
+        self.exptime_dict = (
+            {  # I don't know how to instantiate this so that I get dict[str: float]
+                "camera": 0.0,
+                "electrometer": 0.0,
+                "fiberspectrograph": 0.0,
+            }
+        )
+        self.delay = 1
+
+        self.ready_for_flats = False
```

Comment:
This should not be a flag in the class. Instead we should have a method that checks that the system is in the correct state for flats. We need to make sure these classes are stateless because we can easily be in a situation where one script is used to prepare for flats and another is used to take the calibrations. Since they are completely different instances, they won't know of each other flags. 

Maybe replace this with an `is_ready_for_flats` async method that will do the check? We can probably leave the implementation for another ticket. So, I recommend creating a ticket for the implementation of this method and here you just create the method stub. Something like this:

``` 
async def is_ready_for_flats(self) -> bool:
    """ Add doctring """
    # TODO (DM-12345): Implement method to check that the system is ready for flats.
    return True 
```